### PR TITLE
feat(eds/store): add eds blockstore cache metrics

### PR DESCRIPTION
**Pull request by walldiss**
Add metrics allowing to see blockstore cache size and eviction pressure

### CODE REVIEW

File: share/eds/store.go [hunk 1]
```diff
@@ -407,8 +407,9 @@ func (s *Store) getCachedAccessor(ctx context.Context, key shard.Key) (*accessor
 	if err != nil && err != errCacheMiss {
 		log.Errorf("unexpected error while reading key from bs cache %s: %s", key, err)
 	}
+	// TODO(@walldiss): need to check if err could happen with non-nil accessor
```
Comment (Wondertan) [inline share/eds/store.go:410]: The errCacheMiss can happen yet returning non-nil accessor, IIRC

File: share/eds/accessor_cache.go [hunk 2]
```diff
@@ -49,10 +58,28 @@ func newBlockstoreCache(cacheSize int) (*blockstoreCache, error) {
...
 			log.Errorf("couldn't close accessor after cache eviction: %s", err)
 		}
+		evictedCounter.Add(context.Background(), 1, metric.WithAttributes(
+			attribute.Bool(failedKey, err != nil)))
 	})
 	if err != nil {
 		return nil, fmt.Errorf("failed to instantiate blockstore cache: %w", err)
 	}
+
+	currSize, err := meter.Int64ObservableGauge("eds_blockstore_cache_size",
```
Comment (Wondertan) [inline share/eds/accessor_cache.go:68]: curr?

**Review by renaynay — COMMENTED** [review]
shouldn't we only enable metrics inside blockstore cache if metrics are enabled for eds store?

File: share/eds/store.go (hunk 1 above)
Comment (walldiss) [reply share/eds/store.go:410]: errCacheMiss means there were no accessor, so no accessor should be returned.

File: share/eds/accessor_cache.go (hunk 2 above)
Comment (walldiss) [reply share/eds/accessor_cache.go:68]: cacheSize was already taken by func param

**Review by Wondertan — DISMISSED** [review]

File: share/eds/accessor_cache.go [hunk 3]
```diff
@@ -117,3 +122,41 @@ func (bc *blockstoreCache) unsafeAdd(
...
+	}
+	_, err = meter.RegisterCallback(callback, cacheSize)
+	if err != nil {
+		return err
+	}
+	bc.metrics = &cacheMetrics{evictedCounter: evictedCounter}
+	return nil
+}
+
+func (m *cacheMetrics) observeEvicted(failed bool) {
```
Comment (renaynay) [inline share/eds/accessor_cache.go:156]: not called anywhere - are you going to integrate in this PR or another one?

File: share/eds/store.go (hunk 1 above)
Comment (Wondertan) [inline share/eds/store.go:410]: Can we resolve this todo before merge?

**Review by renaynay — APPROVED** [review]

**Review by distractedm1nd — APPROVED** [review]

**Pull request merged**